src/structures/idt.rs

```diff
@@ -1328,6 +1329,85 @@ pub enum ExceptionVector {
     Security = 0x1E,
 }
 
+/// Exception vector number is invalid
+///
+/// Explains the reason an exception vector number is considered invalid
+#[derive(Debug)]
+pub enum InvalidExceptionVectorNumber {
+    /// Coprocessor Segment Overrun exception is handled as part of the General Protection Fault
+    IsCoprocessorSegmentOverrun(u8),
+    /// Reserved
+    Reserved(u8),
+    /// Only the first 32 numbers can be exception numbers
+    NotException(u8),
+}
+
+impl fmt::Display for InvalidExceptionVectorNumber {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            InvalidExceptionVectorNumber::IsCoprocessorSegmentOverrun(exception_vector_number) => {
+                write!(f, "{exception_vector_number} cannot be a valid exception vector number because the Coprocessor Segment Overrun exception is handled as part of the General Protection Fault")
+            }
+            InvalidExceptionVectorNumber::Reserved(exception_vector_number) => {
+                write!(f, "{exception_vector_number} cannot be a valid exception vector number because it is reserved")
+            }
+            InvalidExceptionVectorNumber::NotException(exception_vector_number) => {
+                write!(f, "{exception_vector_number} cannot be a valid exception vector number because only the first 32 numbers can be exception numbers")
+            }
+        }
+    }
+}
```

Comment:
This is certainly a matter of opinion, but the error messages seem too lengthy to me. Let's just print `"{} is not a valid exception vector"` in all cases and let's also use a struct instead of an enum. It shouldn't be difficult for the user to figure out what went wrong in each case.